internal/xcrd/crd.go

```diff
@@ -97,6 +101,117 @@ func ForCompositeResource(xrd *v1.CompositeResourceDefinition) (*extv1.CustomRes
 	return crd, nil
 }
 
+func ModifyApplyConfigurationForCompositeResource(crd *appcfgextv1.CustomResourceDefinitionApplyConfiguration, xrd *v1.CompositeResourceDefinition) error { //nolint:gocognit
```

Comment:
Needing to do all of this feels a little like it defeats the point of SSA (i.e. just form a simple intent and send it). I wonder if it would be simpler to just do a get-mutate-update of the CRD? I think we could assume exclusive ownership of the CRD spec (i.e. just overwrite that each time).